instant-xml-macros/src/lib.rs

```diff
@@ -55,17 +65,29 @@ struct FieldMeta {
 }
 
 impl FieldMeta {
-    fn from_field(input: &syn::Field) -> FieldMeta {
+    fn from_field(input: &syn::Field) -> Result<FieldMeta, syn::Error> {
         let mut meta = FieldMeta::default();
         for item in meta_items(&input.attrs) {
             match item {
                 MetaItem::Attribute => meta.attribute = true,
                 MetaItem::Ns(ns) => meta.ns = ns,
                 MetaItem::Rename(lit) => meta.rename = Some(lit),
-                MetaItem::Scalar => panic!("attribute 'scalar' invalid in field xml attribute"),
+                MetaItem::RenameAll(_) => {
+                    return Err(syn::Error::new(
+                        input.span(),
```

Comment:
We should use the `item.span()` here.

Reply:
We can `span()` on the `Literal` that comes through, but not on the `item` `MetaItem` instance, which is disconnected from the `syn` types that implement `Spanned`.

Reply:
Okay, I guess doing it on the `Literal` wouldn't be obviously better. Can we do the `input.attrs` instead?

Reply:
I think the way to do this would be to emit a `Vec<(MetaItem, Span)>` from the `meta_items` function.

Reply:
Yeah, but let's leave that for a separate PR.